internal/gitstream/assign.go

```diff
@@ -108,14 +109,33 @@ func (a *Assign) assignIssues(ctx context.Context) error {
 
 			logger.Info("Assigning issue")
 
-			user, err := a.UserHelper.GetUser(ctx, upstreamCommit)
-			if err != nil {
-				return fmt.Errorf("could not get the upstream commit author for downstream issue %d: %v", *issue.Number, err)
-			}
-
-			assignee, err := owners.getAssignee(ctx, a.GC, *user.Login)
-			if err != nil {
-				return fmt.Errorf("could not get select a suitable assignee: %v", err)
+			var (
+				assignee string
+				intErr   error
+			)
+			if user, err := a.UserHelper.GetUser(ctx, upstreamCommit); err != nil {
+				if errors.Is(err, gh.APIErr) {
+					logger.Info("WARNING: Github API is unavailable, skipping commit", "issue", *issue.Number)
+					continue
+				} else {
+					logger.Info("WARNING: commit author for downstream issue not found on github, picking a random assignee",
+						"issue", *issue.Number, "error", err.Error())
+					assignee, intErr = owners.getRandom()
+					if intErr != nil {
+						return fmt.Errorf("could not get a random owner: %v", err)
+					}
+				}
+			} else {
```

Comment:
I think the `else` is not needed here either.

Reply:
I don't think it is. If we get a non API error we will set `assignee` to a random user. In that case we do not want to execute the rest of the code in the `else` block.

Reply:
Added it anyway.